### PR TITLE
Spotcast fix: Remove uneeded SET_CREDENTIALS steps

### PR DESCRIPTION
**Pull request by kylepotts**
This is an attempt to fix https://github.com/home-assistant-libs/pychromecast/issues/520

I noticed that we were no longer receiving a setCredentialsResponse so I decided to try and remove that step and it seemed to work. 

Open to a stronger fix, but this seems to work for me.

### CODE REVIEW

**Comment by Electronlibre2012** [comment]
hello @kylepotts ,

nice! how can i do with hassOs please? i dont think i have access to spotify.py...

need to remove the lines in red? and add the line in green? right?

thanks

**Comment by kylepotts** [comment]
@Electronlibre2012 

I'm not familiar with how things are packaged in hassOs. I currently use a docker container running homeassisant so I can manually patch this file. My guess would be that this would need to be patched and an update to hassOs would need to be pushed out.

**Comment by kazigk** [comment]
I can confirm this fixes #520 for me.

**Comment by fondberg** [comment]
#509 tried the same. Can you check the diff against that one? It seems my speaker also fails

File: pychromecast/controllers/spotify.py [hunk 1]
```diff
@@ -38,16 +40,34 @@ def receive_message(self, _message, data: dict):
...
+            self.client = data["payload"]["clientID"]
+            headers = {
+                'authority': 'spclient.wg.spotify.com',
+                'authorization': 'Bearer {}'.format(self.access_token),
+                'content-type': 'text/plain;charset=UTF-8'
+            }
+
+            request_body = json.dumps({'clientId': self.client, 'deviceId': self.device})
+
+            response = requests.post('https://spclient.wg.spotify.com/device-auth/v1/refresh', headers=headers, data=request_body)
```
Comment (fondberg) [inline pychromecast/controllers/spotify.py:54]: Can you test #509 which doesn't use private APIs to see if it works?

**Review by fondberg — CHANGES_REQUESTED** [review]

File: pychromecast/controllers/spotify.py (hunk 1 above)
Comment (kylepotts) [reply pychromecast/controllers/spotify.py:54]: @fondberg 

The main problem I see is that the authToken used in the addUser request is not the same authToken that is retrieved via the sp_dc and sp_key values. 

I figured out calling that api from looking at whats happening in the browser. If I just use the some authToken thats passed in I actually get a addUserError response.

Comment (kylepotts) [reply pychromecast/controllers/spotify.py:54]: I just tried the code in #509 and I get a timeout.

```
home-assistant   | 2021-08-15 11:31:52 ERROR (MainThread) [homeassistant.core] Error executing service: <ServiceCall spotcast.start (c:ab17c97250473019a4921369e4715140): uri=spotify:playlist:37i9dQZF1DXaa8UmWJHYTU?si=dbed7b69bfdb4e6c, entity_id=media_player.bedroom_speaker, force_playback=False, offset=0, ignore_fully_played=False, shuffle=False, start_volume=101, random_song=False, repeat=off>
home-assistant   | Traceback (most recent call last):
home-assistant   |   File "/usr/src/homeassistant/homeassistant/core.py", line 1507, in catch_exceptions
home-assistant   |     await coro_or_task
home-assistant   |   File "/usr/src/homeassistant/homeassistant/core.py", line 1530, in _execute_service
home-assistant   |     await self._hass.async_add_executor_job(handler.job.target, service_call)
home-assistant   |   File "/usr/local/lib/python3.9/concurrent/futures/thread.py", line 52, in run
home-assistant   |     result = self.fn(*self.args, **self.kwargs)
home-assistant   |   File "/config/custom_components/spotcast/__init__.py", line 143, in start_casting
home-assistant   |     spotify_device_id = spotcast_controller.get_spotify_device_id(
home-assistant   |   File "/config/custom_components/spotcast/spotcast_controller.py", line 202, in get_spotify_device_id
home-assistant   |     spotify_cast_device.startSpotifyController(access_token, expires)
home-assistant   |   File "/config/custom_components/spotcast/spotcast_controller.py", line 84, in startSpotifyController
home-assistant   |     sp.launch_app()
home-assistant   |   File "/usr/local/lib/python3.9/site-packages/pychromecast/controllers/spotify.py", line 87, in launch_app
home-assistant   |     raise LaunchError(
home-assistant   | pychromecast.error.LaunchError: Timeout when waiting for status response from Spotify app

```

Comment (kylepotts) [reply pychromecast/controllers/spotify.py:54]: Trying the code in #509 

Gives me an addUserError

```
home-assistant   | 2021-08-15 11:40:14 DEBUG (Thread-7) [pychromecast.socket_client] [Bedroom speaker(192.168.0.162):8009] Received: Message urn:x-cast:com.spotify.chromecast.secure.v1 from f59a660b-975b-4f40-9c55-e61d0fa040da to sender-0: {'type': 'addUserError', 'payload': {'status': 202}}
home-assistant   | 2021-08-15 11:40:14 ERROR (MainThread) [homeassistant.components.script.j_lo_fi] J Lo-Fi: Error executing script. Unexpected error for call_service at pos 1: Timeout when waiting for status response from Spotify app
home-assistant   | Traceback (most recent call last):
home-assistant   |   File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 371, in _async_step
home-assistant   |     await getattr(self, handler)()
home-assistant   |   File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 571, in _async_call_service_step
home-assistant   |     await service_task
home-assistant   |   File "/usr/src/homeassistant/homeassistant/core.py", line 1491, in async_call
home-assistant   |     task.result()
home-assistant   |   File "/usr/src/homeassistant/homeassistant/core.py", line 1530, in _execute_service
home-assistant   |     await self._hass.async_add_executor_job(handler.job.target, service_call)
home-assistant   |   File "/usr/local/lib/python3.9/concurrent/futures/thread.py", line 52, in run
home-assistant   |     result = self.fn(*self.args, **self.kwargs)
home-assistant   |   File "/config/custom_components/spotcast/__init__.py", line 143, in start_casting
home-assistant   |     spotify_device_id = spotcast_controller.get_spotify_device_id(
home-assistant   |   File "/config/custom_components/spotcast/spotcast_controller.py", line 202, in get_spotify_device_id
home-assistant   |     spotify_cast_device.startSpotifyController(access_token, expires)
home-assistant   |   File "/config/custom_components/spotcast/spotcast_controller.py", line 84, in startSpotifyController
home-assistant   |     sp.launch_app()
home-assistant   |   File "/usr/local/lib/python3.9/site-packages/pychromecast/controllers/spotify.py", line 87, in launch_app
home-assistant   |     raise LaunchError(
home-assistant   | pychromecast.error.LaunchError: Timeout when waiting for status response from Spotify app
home-assistant   | 2021-08-15 11:40:14 ERROR (MainThread) [homeassistant.components.websocket_api.http.connection] [139961477371360] Timeout when waiting for status response from Spotify app
home-assistant   | Traceback (most recent call last):
home-assistant   |   File "/usr/src/homeassistant/homeassistant/components/websocket_api/commands.py", line 185, in handle_call_service
home-assistant   |     await hass.services.async_call(
home-assistant   |   File "/usr/src/homeassistant/homeassistant/core.py", line 1491, in async_call
home-assistant   |     task.result()
home-assistant   |   File "/usr/src/homeassistant/homeassistant/core.py", line 1526, in _execute_service
home-assistant   |     await handler.job.target(service_call)
home-assistant   |   File "/usr/src/homeassistant/homeassistant/components/script/__init__.py", line 278, in service_handler
home-assistant   |     await script_entity.async_turn_on(
home-assistant   |   File "/usr/src/homeassistant/homeassistant/components/script/__init__.py", line 382, in async_turn_on
home-assistant   |     await coro
home-assistant   |   File "/usr/src/homeassistant/homeassistant/components/script/__init__.py", line 409, in _async_run
home-assistant   |     return await self.script.async_run(script_vars, context)
home-assistant   |   File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 1219, in async_run
home-assistant   |     await asyncio.shield(run.async_run())
home-assistant   |   File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 353, in async_run
home-assistant   |     await self._async_step(log_exceptions=False)
home-assistant   |   File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 371, in _async_step
home-assistant   |     await getattr(self, handler)()
home-assistant   |   File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 571, in _async_call_service_step
home-assistant   |     await service_task
home-assistant   |   File "/usr/src/homeassistant/homeassistant/core.py", line 1491, in async_call
home-assistant   |     task.result()
home-assistant   |   File "/usr/src/homeassistant/homeassistant/core.py", line 1530, in _execute_service
home-assistant   |     await self._hass.async_add_executor_job(handler.job.target, service_call)
home-assistant   |   File "/usr/local/lib/python3.9/concurrent/futures/thread.py", line 52, in run
home-assistant   |     result = self.fn(*self.args, **self.kwargs)
home-assistant   |   File "/config/custom_components/spotcast/__init__.py", line 143, in start_casting
home-assistant   |     spotify_device_id = spotcast_controller.get_spotify_device_id(
home-assistant   |   File "/config/custom_components/spotcast/spotcast_controller.py", line 202, in get_spotify_device_id
home-assistant   |     spotify_cast_device.startSpotifyController(access_token, expires)
home-assistant   |   File "/config/custom_components/spotcast/spotcast_controller.py", line 84, in startSpotifyController
home-assistant   |     sp.launch_app()
home-assistant   |   File "/usr/local/lib/python3.9/site-packages/pychromecast/controllers/spotify.py", line 87, in launch_app
home-assistant   |     raise LaunchError(
home-assistant   | pychromecast.error.LaunchError: Timeout when waiting for status response from Spotify app
```

**Comment by fondberg** [comment]
I see . They complicate things a couple of times per year. 
I wonder if they are finished with the onfuscation though.


Also I think spotify controller should probably be liften out from pychromecast as the api is so bluntly closed

**Comment by kylepotts** [comment]
Would it make sense to break out the controller into spotcast instead? I'm not sure if anything internal in HA relies on that controller or not.

**Comment by fondberg** [comment]
It would make sense and would simplify life for @emontnemery  and the gang.
I have very limited time right now due to massive amount of work plus that I'm fed up with trying to find workarounds for spotidy closing their APIs. So if you could make a stab at it it would be brilliant

**Comment by kylepotts** [comment]
Yeah I can take a stab at it. Probably won't be able to work on it till this weekend though.

**Comment by kylepotts** [comment]
Closing in favor of https://github.com/fondberg/spotcast/pull/244 which moves the controller to that repo instead. 

@fondberg take a look when you have a second if you don't mind.

**Pull request closed without merging**